src/glfw/window.cc

```diff
@@ -40,150 +40,113 @@ setWindowHints()
-    throw Error{ error_code, description };
-} // errorCallback
 
-} // namespace
-
-WindowType*
-Window::createWindow(
+WindowHandle*
+WindowImpl::createWindow(
     const WindowConfig& config,
-    Window* bound_handle //
+    WindowImpl* bound_handle //
 )
 {
     setWindowHints();
 
-    WindowType* glfwWindow = glfwCreateWindow(
+    WindowHandle* glfw_window = glfwCreateWindow(
         config.width,
         config.height,
         config.title.c_str(),
         config.fullscreen ? glfwGetPrimaryMonitor() : nullptr,
         nullptr );
 
-    glfwSetWindowUserPointer( glfwWindow, bound_handle );
-    glfwSetFramebufferSizeCallback( glfwWindow, framebufferSizeCallback );
+    glfwSetWindowUserPointer( glfw_window, bound_handle );
+    glfwSetFramebufferSizeCallback( glfw_window, framebufferSizeCallback );
 
-    logGLFWaction(
+    glfw::detail::logAction(
         "Create Window",
-        bound_handle //
+        glfw_window //
     );
 
-    return glfwWindow;
+    return glfw_window;
 } // createWindow
 
 void
-Window::destroyWindow( WindowType* glfwWindow )
+WindowImpl::destroyWindow( WindowHandle* glfw_window )
 {
-    Window* window = getHandle( glfwWindow );
-
-    glfwDestroyWindow( glfwWindow );
+    glfwDestroyWindow( glfw_window );
 
-    logGLFWaction(
+    glfw::detail::logAction(
         "Destroy Window",
-        window //
+        glfw_window //
     );
 } // destroyWindow
 
 void
-Window::framebufferSizeCallback(
-    WindowType* glfwWindow,
+WindowImpl::framebufferSizeCallback(
+    WindowHandle* glfw_window,
     int width,
     int height //
 )
 {
-    Window* window = getHandle( glfwWindow );
+    WindowImpl& window = getHandle( glfw_window );
 
-    logGLFWaction(
+    ::glfw::detail::logAction(
         "Framebuffer resize",
-        window,
-        { fmt::format( "Width = {}", width ), fmt::format( "Height = ", height ) } //
+        glfw_window,
+        { fmt::format( "Width = {}", width ), fmt::format( "Height = {}", height ) } //
     );
 
-    window->m_resize_callback( window, width, height );
+    window.m_resize_callback( width, height );
 } // framebufferSizeCallback
 
 FramebufferSizeType
-Window::getFramebufferSize() const
+WindowImpl::getFramebufferSize() const
 {
     FramebufferSizeType size;
     glfwGetFramebufferSize( m_handle.get(), &size.width, &size.height );
     return size;
 } // getFramebufferSize
 
 void
-Window::setWindowed() const
+WindowImpl::setWindowed()
 {
+    auto [ width, height ] = getFramebufferSize();
+
     glfwSetWindowMonitor(
         m_handle.get(),
         nullptr,
         0,
         0,
-        getFramebufferWidth(),
-        getFramebufferHeight(),
-        GLFW_DONT_CARE );
-
-    logGLFWaction(
-        "Set windowed mode",
-        this //
+        width,
+        height,
+        GLFW_DONT_CARE //
     );
+
+    ::glfw::detail::logAction( "Set windowed mode", m_handle.get() );
 } // setWindowed
 
 void
-Window::setFullscreen() const
+WindowImpl::setFullscreen()
```

Comment:
Const method.

Reply:
It's not really semantically const, because it changes the state of the window

Reply:
Previously I thought about the constancy of an object than about semantic constancy. Probably, it makes sense.